transformers/events/DOCUMENTATION.md

```diff
@@ -2,116 +2,206 @@
+1. For each event in the contract:
+    1. Write a test for the event signature in 
+    [`signature_test.go`](../shared/constants/signature_test.go)
+        1. To find out the event's signature you can use this [Keccak-256
+       calculator](https://emn178.github.io/online-tools/keccak_256.html). For Example:
+            * Search Etherscan for the contract `0x39755357759cE0d7f32dC8dC45414CCa409AE24e`. 
+            * Open the 'Contract' tab and in the contract source code and search
+            for the `LogItemUpdate` event. It looks like this: `event LogItemUpdate(uint id);`. 
+            * You can copy that string and paste it into the Keccak-256
+            calculator (link above). Remove the parameter names and use
+            `uint256` for the `unit` so like this: `LogItemUpdate(uint256)`.
+            This will output an event signature.
+            * To verify you got the signature right you can take that signature
+            (in this case
+            `a2c251311b1a7a475913900a2a73dc9789a21b04bc737e050bbc506dd4eb3488`) 
+            and search for it on the events tab of the contract. Make sure you 
+            prefix it with `0x`. The results should include the event you are
+            looking for, provided one already exists for it.
+    1. Make the test pass by updating the function list in `signature.go` as
+       well as updating `method.go` as needed. Both of these files are in the
+       constants package.
+    1. Create test event in [`test_data`](../test_data)
+        * Use Data from a real event in Etherscan wherever possible.
+          Particularly when it comes to Topics and the Data entries. You'll
+          thank me later.
+        * Update constants as needed. Look at older examples of test data for
+          inspiration.
+        * You do not need a database migration at this point.
+    1. Create a new directory in `events` named after your new transformer. This
+       will be the package your transformer is stored in.
+    1. Create three files - `transformer.go`, `transformer_test.go` and
+       `<package_name>_test.go` in the new directory. See the other transformers
+       for examples.
+    1. `<package_name>_test.go` lets `gingko` run tests in this directory. The
+       file should look something like this:
+       
+```go
+ package <package_name>_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestLogMinSell(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "<PackageName>  Suite")
+}
+```
+   1. Implement the toModels function in `transformer.go` with the appropriate
+   testing in `transformer_test.go` of course. This means converting the raw log 
+   into a go struct. 
+   1. For custom events you can convert each EventLog entry into an entity
+   using the function `contract.UnpackLog(&entity, "<EventLogName>", log.Log)`.
+   The entity is usually defined (by you) in a file called `entity.go` in the
```

Comment:
"by you" is probably the right answer here although in theory `abigen` may be helpful/necessary if figuring out the type mappings is challenging

Reply:
I added a note about `abigen`